pkg/rbac/parser.go

```diff
@@ -68,20 +117,33 @@ type Generator struct {
 func (Generator) RegisterMarkers(into *markers.Registry) error {
 	return into.Register(RuleDefinition)
 }
+
 func (g Generator) Generate(ctx *genall.GenerationContext) error {
-	var rules []rbacv1.PolicyRule
+	rules := make(map[ruleKey]*Rule)
 	for _, root := range ctx.Roots {
 		markerSet, err := markers.PackageMarkers(ctx.Collector, root)
 		if err != nil {
 			root.AddError(err)
 		}
 
-		for _, rule := range markerSet[RuleDefinition.Name] {
-			rules = append(rules, rule.(Rule).ToRule())
+		for _, markerValues := range markerSet[RuleDefinition.Name] {
+			rule := markerValues.(Rule)
+			key := rule.key()
+			if _, ok := rules[key]; !ok {
```

Comment:
this would be a lot clearer as `rules[rule.key()] = append(rules[rule.key()], rule.Verbs)`, and then dedup later.

Reply:
`Append in the first round + dedup in a second round` does have one issue: in the case when some marker values have lots of duplicates, the rule values may have lots of dups.

The current method is clear to me in that it only needs one round to get things done.

Reply:
¯\_(ツ)_/¯

I suppose it's fine either way, but to your point: this method creates a lot more intermediate objects.  In either case, the concession is relatively small, since this probably isn't a hot path.

Reply:
Thanks for sharing your insight on this! Then I would leave this as it is.